### PR TITLE
Add support for increasing the number of shards

### PR DESCRIPTION
**Pull request by mfussenegger**
- [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

**Comment by mfussenegger** [comment]
@smakalias @seut pushed another commit to make the rename+drop atomic

File: sql/src/main/java/io/crate/execution/ddl/index/SwapAndDropIndexRequest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+/**
+ * Renames source to target, and dropping the former target - effectively overriding the target
+ */
+public final class SwapAndDropIndexRequest extends AcknowledgedRequest<SwapAndDropIndexRequest> {
+
+    private String source;
```
Comment (smakalias) [inline sql/src/main/java/io/crate/execution/ddl/index/SwapAndDropIndexRequest.java]: minor: could be final

File: sql/src/main/java/io/crate/execution/ddl/index/TransportSwapAndDropIndexNameAction.java [hunk 2]
```diff
@@ -39,39 +39,39 @@
...
+                                               TransportService transportService,
+                                               ClusterService clusterService,
+                                               ThreadPool threadPool,
+                                               ActionFilters actionFilters,
+                                               AllocationService allocationService,
+                                               IndexNameExpressionResolver indexNameExpressionResolver) {
         super(settings, ACTION_NAME, transportService, clusterService, threadPool, actionFilters,
-            indexNameExpressionResolver, BulkRenameIndexRequest::new, AcknowledgedResponse::new, AcknowledgedResponse::new,
+            indexNameExpressionResolver, SwapAndDropIndexRequest::new, AcknowledgedResponse::new, AcknowledgedResponse::new,
             "bulk-rename-index");
```
Comment (smakalias) [inline sql/src/main/java/io/crate/execution/ddl/index/TransportSwapAndDropIndexNameAction.java]: pls rename this also

File: sql/src/test/java/io/crate/integrationtests/ResizeShardsITest.java [hunk 3]
```diff
@@ -60,24 +60,20 @@ public void testShrinkShardsEnsureLeftOverIndicesAreRemoved() throws Exception {
...
         ensureYellow();
 
-        final String targetIndexName = ".resized." + getFqn("quotes");
-        final String backupIndexName = ".backup." + getFqn("quotes");
-        logger.info("targetIndexName:" + targetIndexName );
-
-        createIndex(targetIndexName, backupIndexName);
+        final String resizeIndex = ".resized." + getFqn("quotes");
+        createIndex(resizeIndex);
+        logger.info("targetIndexName:" + resizeIndex );
```
Comment (smakalias) [inline sql/src/test/java/io/crate/integrationtests/ResizeShardsITest.java]: could you please remove that? 

I forgot to do so when I fixed the test to work for randomised schema. 
thanks

File: sql/src/main/java/io/crate/metadata/cluster/SwapAndDropIndexExecutor.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        routingBuilder.addAsFromCloseToOpen(newIndexMetaData);
+        blocksBuilder.addBlocks(newIndexMetaData);
+
+        return allocationService.reroute(
+            ClusterState.builder(currentState)
+                .metaData(mdBuilder)
+                .routingTable(routingBuilder.build())
+                .blocks(blocksBuilder)
+                .build(),
+            "indices bulk rename"
```
Comment (smakalias) [inline sql/src/main/java/io/crate/metadata/cluster/SwapAndDropIndexExecutor.java]: pls also rename

File: sql/src/main/java/io/crate/execution/ddl/index/SwapAndDropIndexRequest.java (hunk 1 above)
Comment (mfussenegger) [reply sql/src/main/java/io/crate/execution/ddl/index/SwapAndDropIndexRequest.java]: It can't be, we need the empty constructor for the transport action.

Comment (smakalias) [reply sql/src/main/java/io/crate/execution/ddl/index/SwapAndDropIndexRequest.java]: ah, you are right. Missed that.

**Pull request merged**